BezierKit/Library/PathComponent+Arc.swift

```diff
@@ -0,0 +1,166 @@
+            .scaledBy(
+                x: rect.width / diameter,
+                y: rect.height / diameter
+            )
+        
+        self.init(
+            points: tempPath.points.map {
+                $0.applying(transform)
+            },
+            orders: tempPath.orders
+        )
+    }
+    
+    public convenience init(
+        arcCenter center: CGPoint,
+        radius: CGFloat,
+        startAngle: CGFloat,
+        endAngle: CGFloat,
+        clockwise: Bool,
+        numberOfCurves: Int = 4
+    ) {
+        precondition(numberOfCurves > 0)
+        
+        var curves = [CubicCurve]()
+        let distance = Self.distanceBetween(
+            startAngle,
+            endAngle,
+            clockwise: clockwise
+        )
+        let unitRadians = 2 * .pi / CGFloat(numberOfCurves)
+        let numberOfFullUnits = Int(distance / unitRadians)
+        let isClosed = numberOfFullUnits == numberOfCurves
+        var currentPoint = CGPoint.zero
+        
+        func addArc(
+            unitRadians: CGFloat,
+            rotateRadians: CGFloat,
+            endPoint: CGPoint? = nil
+        ) {
+            let transform = CGAffineTransform
+                .init(scaleX: radius, y: radius)
+                .rotated(by: rotateRadians)
+            let cp1 = Self.controlPoint1(unitRadians)
+                .applying(transform)
+            let cp2 = Self.controlPoint2(unitRadians)
+                .applying(transform)
+            let endPoint = endPoint ?? (CGPoint(x: cos(unitRadians), y: sin(unitRadians))
+                .applying(transform))
+            
+            let curve = CubicCurve(
+                p0: currentPoint, p1: cp1,
+                p2: cp2, p3: endPoint
+            )
+            curves.append(curve)
+            currentPoint = endPoint
+        }
+        
+        currentPoint = CGPoint(x: radius, y: 0)
+        for i in 0 ..< numberOfFullUnits {
+            var endPoint: CGPoint?
+            if isClosed && i == numberOfFullUnits - 1 {
+                endPoint = CGPoint(x: radius, y: 0)
+            }
+            addArc(
+                unitRadians: unitRadians,
+                rotateRadians: unitRadians * CGFloat(i),
+                endPoint: endPoint
+            )
+        }
+        
+        if !isClosed {
+            let rotateRadians = unitRadians * CGFloat(numberOfFullUnits)
+            addArc(
+                unitRadians: distance - rotateRadians,
+                rotateRadians: rotateRadians
+            )
+        }
+        
+        var finalTransform = CGAffineTransform.identity
+        finalTransform = finalTransform
+            .translatedBy(x: center.x, y: center.y)
+            .rotated(by: startAngle)
+        if !clockwise {
+            finalTransform = finalTransform
+                .scaledBy(x: 1, y: -1)
+        }
+        curves = curves.map {
+            CubicCurve(points: $0.points.map {
+                $0.applying(finalTransform)
+            })
```

Comment:
use `copy(using: CGAffineTransform) -> Self` instead of transforming the array of points.